catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/PenColorBrickTest.java

```diff
@@ -0,0 +1,225 @@
+				.perform(CustomSwipeAction.swipeLeftSlow());
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(pressBack());
+		onView(withText(R.string.formula_editor_discard_changes_dialog_title))
+				.check(matches(isDisplayed()));
+		onView(withId(android.R.id.button1))
+				.perform(click());
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text)
+				.checkShowsNumber(rgbValueRed);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPenColorBrickBlue() {
+		int positionPenColorBrick = 1;
+		int rgbValueBlue = 3;
+
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_blue_edit_text)
+				.perform(click());
+		onView(withId(R.id.color_rgb_preview))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.color_rgb_seekbar_blue))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.rgb_blue_value))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.color_rgb_textview_blue))
+				.check(matches(isDisplayed()));
+
+		onView(withId(R.id.rgb_blue_value))
+				.perform(click());
+		onFormulaEditor()
+				.performEnterNumber(rgbValueBlue);
+		onFormulaEditor()
+				.performCloseAndSave();
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_blue_edit_text)
+				.checkShowsNumber(rgbValueBlue);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPenColorBrickGreen() {
+		int positionPenColorBrick = 1;
+		int rgbValueGreen = 255;
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_green_edit_text)
+				.perform(click());
+		onView(withId(R.id.color_rgb_seekbar_green))
+				.perform(CustomSwipeAction.swipeRightSlow());
+		onView(withId(R.id.color_rgb_seekbar_green))
+				.perform(pressBack());
+		onView(withText(R.string.formula_editor_discard_changes_dialog_title))
+				.check(matches(isDisplayed()));
+		onView(withId(android.R.id.button1))
+				.perform(click());
+
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_green_edit_text)
+				.checkShowsNumber(rgbValueGreen);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testPenColorBrickUndoRedo() {
+		int positionPenColorBrick = 1;
+		float[] setRgbValuesRed = {36, 45, 98};
+		float[] setRgbValuesGreen = {122, 24, 0};
+		float[] setRgbValuesBlue = {44, 255};
+
+		onBrickAtPosition(positionPenColorBrick).onFormulaTextField(R.id.brick_set_pen_color_action_red_edit_text)
+				.perform(click());
+		onView(withId(R.id.color_rgb_seekbar_red))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesRed[0] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_green))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesGreen[0] / 255.0f));
+		onView(withId(R.id.color_rgb_seekbar_blue))
+				.perform(CustomSwipeAction.swipeToPosition(setRgbValuesBlue[0] / 255.0f));
```

Comment:
please wrap all the color picker fragment stuff in a viewInteractionWrapper
have a look at FormulaEditorWrapper